### PR TITLE
Ignore permission error on reading configuration file in ${HOME}/.minio

### LINKED ISSUE #8163: minIO server writes to deprecated `$HOME/.minio`
## Expected Behavior
minIO server shouldn't write to `$HOME/.minio` since the `--config-dir` is deprecated.

## Current Behavior
On server startup minIO tries to write to `$HOME/.minio/config.json`. If it can't it exits with the following message.

```
ERROR Unable to initialize config system: Insufficient permissions to access path, `/home/minio/.minio/config.json`
```

If `$HOME/.minio` isn't writable, this forces the deprecated `--config-dir` flag to be used.

## Steps to Reproduce
1. `useradd -M -r minio`
2. `su minio`
3. `minio server data`

## Your Environment
* Version used (`minio version`):
```
Version: 2019-08-29T00:25:01Z
Release-Tag: RELEASE.2019-08-29T00-25-01Z
Commit-ID: eb18c82976251d9a0c0730df27e10204c89beae2
```
* Operating System and version (`uname -a`):
```
Linux 5.2.9-200.fc30.x86_64
```

### PR DESCRIPTION
**Pull request by balamurugana**
Fixes: #8163

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

File: cmd/config-migrate.go [hunk 1]
```diff
@@ -68,6 +69,9 @@ func migrateConfig() error {
 		if os.IsNotExist(err) {
 			return nil
 		}
+		if errors.Is(err, os.ErrPermission) {
```
Comment (harshavardhana) [inline cmd/config-migrate.go:72]: There is already os.IsPermission(err) available which can be used...

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#4885be6e-445c-4988-b54f-b2401d39733f) |
| mint-dist-xl.sh | :x: [more...](#9ac6c038-6cc7-480a-b64c-970255de6517) |
<a name="9ac6c038-6cc7-480a-b64c-970255de6517"></a>
### 8315-09a0c24/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:32355
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a56652cdcb5d:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 1 minutes and 16 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 52 seconds
(8/14) Running minio-dotnet tests ... done in 39 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 37 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 28 seconds
(11/14) Running minio-js tests ... done in 51 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 7 seconds
{
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "message": "'content-language'",
  "status": "FAIL",
  "name": "minio-py:test_presigned_get_object_response_headers",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1396, in test_presigned_get_object_response_headers\n    returned_content_language = response.headers['Content-Language']\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/_collections.py\", line 156, in __getitem__\n    val = self._container[key.lower()]\nKeyError: 'content-language'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2039, in main\n    test_presigned_get_object_response_headers(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1414, in test_presigned_get_object_response_headers\n    raise Exception(err)\nException: 'content-language'\n",
  "duration": 253,
  "args": {
    "presigned_get_object_url": "http://72.28.97.51:32355/minio-py-test-27083f9b-70a7-4df4-b3de-72712eb4630b/e1d9f45a-ee02-4b75-93ee-f0ff0b56e5b3?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20190926%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20190926T025545Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&response-content-language=en_US&response-content-type=text%2Fplain&X-Amz-Signature=e297a6f85cb397706531e1bf856cb45e873e50630a8173cbe37aa309b36844a4",
    "bucket_name": "minio-py-test-27083f9b-70a7-4df4-b3de-72712eb4630b",
    "object_name": "e1d9f45a-ee02-4b75-93ee-f0ff0b56e5b3"
  }
}

Executed 11 out of 14 tests successfully.
```
<a name="4885be6e-445c-4988-b54f-b2401d39733f"></a>
### 8315-09a0c24/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:31661
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1882b561c869:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 53 seconds
(4/14) Running aws-sdk-ruby tests ... done in 15 seconds
(5/14) Running awscli tests ... done in 2 minutes and 35 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 1 minutes and 11 seconds
(8/14) Running minio-dotnet tests ... done in 2 minutes and 1 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 39 seconds
{
  "args": {},
  "duration": 76876,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```

File: cmd/config-migrate.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/config-migrate.go:72]: @balamurugana ^^

**Comment by harshavardhana** [comment]
Closing this as fixed already in master and released.

**Pull request closed without merging**